tasks/Task.cpp

```diff
@@ -762,11 +763,60 @@ void Task::factoryReset(Arena::IDevice* device, Arena::ISystem& system)
             online = true;
         }
         catch (GenICam::TimeoutException& ex) {
-            if (base::Time::now() - start_time >= _camera_reset_timeout.get()) {
+            if (base::Time::now() - start_time >=
+                _camera_config.get().camera_reset_timeout) {
                 LOG_ERROR_S << "Timeout waiting for camera restart." << endl;
                 throw runtime_error("Timeout waiting for camera restart.");
             }
             usleep(500000);
         }
     }
+}
+
+void Task::collectInfo()
+{
+    auto current_time = base::Time::now();
+
+    if (current_time - m_last_message >= _camera_config.get().update_info) {
```

Comment:
Do an early return.

if (current_time ... < update_info_period) {
   return;
}

... do the rest.